app/components/search/search.tsx

```diff
@@ -11,7 +11,10 @@ const Search = () => {
   const [keywords, setKeywords] = useState("");
   const [publicDomainOnly, setPublicDomainOnly] = useState(false);
 
-  const handleSubmit = (event, router) => {
+  const handleSubmit = (
+    event: React.FormEvent<HTMLFormElement>,
+    router: ReturnType<typeof useRouter>
```

Comment:
first time I've done this with the `ReturnType` but it seems to work. I don't think this is a good thing to use a lot since it's kind of unclear, but there's no type for `AppRouterInstance`

Reply:
Yeah, you'd have to think about what's going on to understand it.

What you can also do is remove `router` as an argument. I don't think I noticed this before but it's not wrong to use the router from the global scope so you don't pass it to the function.